### PR TITLE
CI pipeline tweaking

### PR DESCRIPTION
**Pull request by tebbon**
## Description :sparkles:
Pull request to first test the env variables to get the review environment up, and then to add the correct values to staging as well. DO NOT MERGE YET, plz.

### CODE REVIEW

**Review by santtul — APPROVED** [review]
👍

File: .prod/uwsgi.ini [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+static-map = /static=/var/berth/static
+uid = appuser
+gid = appuser
+buffer-size = 32768
+master = 1
+processes = 2
+threads = 2
+log-format = %(addr) - %(user) [%(ltime)] "%(method) %(uri) %(proto)" %(status) %(size) "%(referer)" "%(uagent)"
+route = /healthz donotlog:
+route = /readiness donotlog:
```
Comment (a-rmz) [inline .prod/uwsgi.ini:14]: Nitpick: newline at the end 😅

File: Dockerfile [hunk 2]
```diff
@@ -28,6 +28,12 @@ RUN apt-install.sh \
 COPY --chown=appuser:appuser docker-entrypoint.sh /app
 ENTRYPOINT ["./docker-entrypoint.sh"]
 
+ENV STATIC_ROOT /var/berth/static
+ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONUNBUFFERED 1
+
+RUN mkdir -p /var/berth/static
+RUN chown appuser:appuser /var/berth/static
```
Comment (a-rmz) [inline Dockerfile:36]: Do we still need/use the `static`s dir? 🤔 (AFAIK we don't)

File: docker-entrypoint.sh [hunk 3]
```diff
@@ -19,6 +19,17 @@ if [[ "$APPLY_MIGRATIONS" = "1" ]]; then
 fi
 
 
-# Start server TODO run with uwsgi like tunnistamo?
-python ./manage.py runserver 0.0.0.0:8000
+if [[ "$CREATE_SUPERUSER" = "1" ]]; then
+python ./manage.py shell -c "from django.contrib.auth import get_user_model; User = get_user_model(); User.objects.filter(email='admin@example.com').delete(); User.objects.create_superuser('admin', 'admin@example.com', 'adminpass')"
```
Comment (a-rmz) [inline docker-entrypoint.sh:23]: Missing indentation

File: docker-entrypoint.sh [hunk 4]
```diff
@@ -19,6 +19,17 @@ if [[ "$APPLY_MIGRATIONS" = "1" ]]; then
...
 
+# Start server
+if [[ ! -z "$@" ]]; then
+    "$@"
+elif [[ "$DEV_SERVER" = "1" ]]; then
+    python ./manage.py runserver 0.0.0.0:8000
+else
+    python ./manage.py collectstatic --noinput
+    uwsgi --ini .prod/uwsgi.ini
+fi
```
Comment (a-rmz) [inline docker-entrypoint.sh:35]: Nitpick: newline

File: docker-entrypoint.sh [hunk 5]
```diff
@@ -19,6 +19,17 @@ if [[ "$APPLY_MIGRATIONS" = "1" ]]; then
...
+ echo "Admin user created with credentials admin:adminpass (email: admin@example.com)"
+fi
 
+# Start server
+if [[ ! -z "$@" ]]; then
+    "$@"
+elif [[ "$DEV_SERVER" = "1" ]]; then
+    python ./manage.py runserver 0.0.0.0:8000
+else
+    python ./manage.py collectstatic --noinput
```
Comment (a-rmz) [inline docker-entrypoint.sh:33]: Now that the images have been moved to the customer UI, we don't need to run this

File: Dockerfile (hunk 2 above)
Comment (a-rmz) [reply Dockerfile:36]: Resolved: required for Django admin

File: docker-entrypoint.sh (hunk 3 above)
Comment (a-rmz) [reply docker-entrypoint.sh:23]: Let's add this as a management command instead.

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -57,21 +58,21 @@ staging:
...
-
     K8S_SECRET_VENE_UI_RETURN_URL: "https://venepaikat.hel.ninja/"
-    K8S_SECRET_VENE_PAYMENTS_BAMBORA_API_URL: "https://real-bambora-api-url/api"
-    K8S_SECRET_VENE_PAYMENTS_BAMBORA_API_KEY: "dummy-key"
-    K8S_SECRET_VENE_PAYMENTS_BAMBORA_API_SECRET: "dummy-secret"
-    K8S_SECRET_VENE_PAYMENTS_BAMBORA_PAYMENT_METHODS: "dummy-bank"
+    K8S_SECRET_VENE_PAYMENTS_BAMBORA_API_URL: "$GL_QA_BAMBORA_API_URL"
+    K8S_SECRET_VENE_PAYMENTS_BAMBORA_API_KEY: "$GL_QA_BAMBORA_API_KEY"
+    K8S_SECRET_VENE_PAYMENTS_BAMBORA_API_SECRET: "$GL_QA_BAMBORA_API_SECRET"
+    K8S_SECRET_VENE_PAYMENTS_BAMBORA_PAYMENT_METHODS: "$GL_QA_BAMBORA_PAYMENT_METHODS"
```
Comment (a-rmz) [inline .gitlab-ci.yml:77]: Probably we'll need the Elastic APM variables as well (let's confirm the names first)
```env
ELASTIC_APM_SERVICE_NAME=<SERVICE-NAME>
ELASTIC_APM_SECRET_TOKEN=<SECRET-TOKEN>
```

File: docker-entrypoint.sh (hunk 3 above)
Comment (tebbon) [reply docker-entrypoint.sh]: doned

File: .gitlab-ci.yml (hunk 6 above)
Comment (tebbon) [reply .gitlab-ci.yml]: staging variables should now be in place, let's test it out

File: users/management/commands/add_admin_user.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        parser.add_argument(
+            "-p", "--password", type=str, help="Password", default="admin"
+        )
+        parser.add_argument(
+            "-e", "--email", type=str, help="Email", default="admin@example.com"
+        )
+
+    def handle(self, *args, **kwargs):
+        user = get_user_model()
+        if not user.objects.filter(username=kwargs["username"]).count():
```
Comment (a-rmz) [inline users/management/commands/add_admin_user.py:21]: For improved readability:
```suggestion
        if not user.objects.filter(username=kwargs["username"]).exists():
```

**Review by a-rmz — APPROVED** [review]

**Pull request merged**